sycl/source/detail/usm/usm_impl.cpp

```diff
@@ -143,182 +166,231 @@ void free(void *Ptr, const context &Ctxt) {
   }
 }
 
+// For ABI compatibility
+__SYCL_EXPORT void *alignedAllocHost(size_t Alignment, size_t Size,
+                                     const context &Ctxt, alloc Kind) {
+  return alignedAllocHost(Alignment, Size, Ctxt, Kind, detail::code_location{});
```

Comment:
Arguable. The idea is that for old ABI we don't really have any location data, so we just create an empty object here instead of some real values. And since this code is in runtime lib, it'll be compiled with `NDEBUG`, and code location will be fused off anyway.